### PR TITLE
scheduled removal of `Trainer.running_sanity_check`

### PR DESCRIPTION
**Pull request by Tshimanga**
## What does this PR do?

Removes deprecated property `Trainer.running_sanity_checking` as scheduled for 1.5

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/deprecated_api.py [hunk 1]
```diff
@@ -17,16 +17,8 @@
...
-    sanity_checking: bool
     fit_loop: FitLoop
 
-    @property
-    def running_sanity_check(self) -> bool:
-        rank_zero_deprecation(
-            "`Trainer.running_sanity_check` has been renamed to `Trainer.sanity_checking` and will be removed in v1.5."
-        )
-        return self.sanity_checking
-
```
Comment (Tshimanga) [inline pytorch_lightning/trainer/deprecated_api.py:29]: I _really_ like this approach to encapsulating deprecated apis. super cool :)

**Review by awaelchli — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -237,6 +237,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Removed deprecated `profiled_functions` argument from `PyTorchProfiler` ([#9178](https://github.com/PyTorchLightning/pytorch-lightning/pull/9178))
 
+
+- Removed deprecated property `Trainer.running_sanity_check` ([#9209](https://github.com/PyTorchLightning/pytorch-lightning/pull/9209))
```
Comment (awaelchli) [inline CHANGELOG.md:241]: let's mention that this is in favor of `Trainer.sanity_checking`

**Comment by awaelchli** [comment]
@Tshimanga thanks for helping out here!

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Comment by radlera** [comment]
Hi there, 
it seems this PR had a breaking change with the ray[tune] [TuneReportCallback](https://github.com/ray-project/ray/blob/ac843a957c6e5f1f7dbafcec79cb4fc2e8b9db3c/python/ray/tune/integration/pytorch_lightning.py#L177).
Are there plans to fix this already?

**Comment by awaelchli** [comment]
@radlera This was renamed to `sanity_checking`.

**Comment by awaelchli** [comment]
@kaushikb11 makes some updates here as well: https://github.com/ray-project/ray/pull/20562

**Pull request merged**